src/crimson/osd/osd_operations/client_request_common.cc

```diff
@@ -73,6 +73,10 @@ CommonClientRequest::do_recover_missing(
       !pg->is_degraded_or_backfilling_object(soid)) {
     return seastar::now();
   }
+  auto &peering_state = pg->get_peering_state();
+  auto &missing_loc = peering_state.get_missing_loc();
+  bool needs_recovery = missing_loc.needs_recovery(soid, &ver);
```

Comment:
We override `ver` from calling `is_unreadable_object`. Do we still need the previous `ver` usage?

Reply:
Can you please cherry-pick 61e883499c6727d21de1ee306c7036915ffd8255 here? logs may be useful here.